core/io/marshalls.cpp

```diff
@@ -59,6 +62,13 @@ ObjectID EncodedObjectAsID::get_object_id() const {
 #define ENCODE_FLAG_64 1 << 16
 #define ENCODE_FLAG_OBJECT_AS_ID 1 << 16
 
+// Actually, these are not flags.
+#define ENCODE_FLAG_TYPED_ARRAY_BUILTIN 1 << 16
+#define ENCODE_FLAG_TYPED_ARRAY_CLASS_NAME 2 << 16
+#define ENCODE_FLAG_TYPED_ARRAY_SCRIPT 3 << 16
+// Occupies bits 16 and 17. If we need more flags, we can place them after.
+#define ENCODE_FLAG_TYPED_ARRAY_MASK 0b11 << 16
```

Comment:
After some thinking, I suggest `ENCODE_FIELD_TYPED_ARRAY_*`.

Regarding parentheses, I realize the existing defined weren't using them. So, for consistency, either ignore my suggestion or add them to the existing ones (me liking the latter better).